### PR TITLE
single device only supports binding a single node

### PR DESCRIPTION
**Pull request by zc2638**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
```
In the actual scenario, no device needs to bind multiple gateways.
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
crd resource need to update
```

### CODE REVIEW

File: cloud/pkg/apis/devices/v1alpha2/device_instance_types.go [hunk 1]
```diff
@@ -38,10 +38,9 @@ type DeviceSpec struct {
 	// on edge node.
 	// +optional
 	Data DeviceData `json:"data,omitempty"`
-	// NodeSelector indicates the binding preferences between devices and nodes.
-	// Refer to k8s.io/kubernetes/pkg/apis/core NodeSelector for more details
+	// NodeName indicates the binding preferences between device and node.
 	// +optional
-	NodeSelector *v1.NodeSelector `json:"nodeSelector,omitempty"`
```
Comment (fisherxu) [inline cloud/pkg/apis/devices/v1alpha2/device_instance_types.go:44]: Now one device only bind to one node, but we need to leave the interface for binding to multi-node for future. We can have more discussion.

Comment (zc2638) [reply cloud/pkg/apis/devices/v1alpha2/device_instance_types.go:44]: Yes, we can discuss it later.

**Comment by subpathdev** [comment]
Like @fisherxu  I think using a NodeSelector is better than setting a NodeName explicit. Can we discuss this in this PR?

File: build/crds/devices/devices_v1alpha2_device.yaml [hunk 2]
```diff
@@ -392,7 +391,6 @@ spec:
               type: object
           required:
             - deviceModelRef
-            - nodeSelector
```
Comment (subpathdev) [inline build/crds/devices/devices_v1alpha2_device.yaml:395]: If we want to use NodeName instead, I would prefer if that is also required.

**Comment by kevin-wangzefeng** [comment]
The nodeselector is a preference here, we don't have an explicit binding indicator so far, which is a problem.
Could you describe more about your use case and scenarios and we can discuss how to improve the API.

**Comment by zc2638** [comment]
> The nodeselector is a preference here, we don't have an explicit binding indicator so far, which is a problem.
> Could you describe more about your use case and scenarios and we can discuss how to improve the API.

So far, the device associated node has only realized a single node, and its purpose is very fixed. It may not be changed for a long time in the future. It is better to use a more concise correlation method.

In terms of scenarios, most of what we see are scenarios from devices to single nodes. I'm not sure if there will be other multi-node scenarios, we can have more discussion here.

**Pull request closed without merging**